### PR TITLE
Preview sept 2018 javascript update

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
This is a new Custom Functions JavaScript branch based on Michael's changes.  Despite the fact that everything builds correctly and seems to load correctly, the custom functions aren't working for me.  Will need to look into that but for now, I want to get these changes pushed and for people to be aware that this work is in progress.

### CODE REVIEW

**Comment by mscharlock** [comment]
Has anyone managed to get CFs working using this new thing? Just curious because if not we should not update IMHO? (i.e. we need the HOL to work at Ignite).

File: manifest.xml [hunk 1]
```diff
@@ -38,9 +38,9 @@
 		</Hosts>
 		<Resources>
 			<bt:Urls>
-				<bt:Url id="JSON-URL" DefaultValue="https://localhost:8081/config/customfunctions.json" />
-				<bt:Url id="JS-URL" DefaultValue="https://localhost:8081/dist/win32/ship/index.win32.bundle" />
-				<bt:Url id="HTML-URL" DefaultValue="https://localhost:8081/index.html" />
+				<bt:Url id="JSON-URL" DefaultValue="https://localhost:8081/src/customfunctions/functions.json" />
+				<bt:Url id="JS-URL" DefaultValue="https://localhost:8081/customfunctions.bundle.js" />
+				<bt:Url id="HTML-URL" DefaultValue="https://localhost:8081/public/about.html" />
```
Comment (akrantz) [inline manifest.xml]: Shouldn't this be customfunctions.html ?

File: src/customfunctions/functions.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import * as MyFunctions from "./sample";
```
Comment (akrantz) [inline src/customfunctions/functions.js]: this should import specific functions, and assign each function to CustomFunctionMappings.ID

File: src/customfunctions/functions.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+// To expose the functions to Excel, we need to set the
+//    CustomFunctionsMapping variable with the set of functions
+CustomFunctionMappings = MyFunctions;
```
Comment (akrantz) [inline src/customfunctions/functions.js]: It makes it unclear if you assign everything exported to CustomFunctionMappings.

**Comment by akrantz** [comment]
Is there also a typescript update?

File: public/customfunctions.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  <meta charset="UTF-8" />
+  <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
+  <meta http-equiv="Expires" content="0" />
+  <title>Custom Functions</title>
+  <script src="https://officedev.github.io/custom-functions/lib/custom-functions-runtime.js"></script>
+  <script src="./../dist/customfunctions.bundle.js"></script>
+</head>
+
+<body>
+  This page is used to run Custom Functions in a hidden browser instance or iframe,
```
Comment (mscharlock) [inline public/customfunctions.html]: Random nit: custom functions should be all lowercase, no caps

File: public/about.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <link rel="stylesheet" href="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-core/9.6.0/css/fabric.min.css">    
+
+    <script>
+        Office.onReady(function() { /* nothing to initialize */ })
+    </script>
+</head>
+
+<body class="ms-Fabric">
+    <p>
+        You can use this sample to try Custom Functions.
```
Comment (mscharlock) [inline public/about.html]: So custom functions does not need caps - both can be lowercase when not in a title.

File: README.md [hunk 6]
```diff
@@ -20,7 +20,7 @@ Learn how to use custom functions in Excel (similar to user-defined functions, o
 
 ## Prerequisites
 
-* Install Office 2016 for Windows and join the [Office Insider](https://products.office.com/en-us/office-insider) program. You must have Office build number 8711 or later.
+* Install Office 2016 for Windows and join the [Office Insider](https://products.office.com/en-us/office-insider) program. You must have Office build number 10827 or later.
```
Comment (mscharlock) [inline README.md]: Link should not include en-us - change to https://products.office.com/office-insider

File: public/about.html [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        Note that this file is only for the instructions; the file used by the <i>runner</i> of the Custom Functions is inside of the "customfunctions/index.html" folder.
```
Comment (mscharlock) [inline public/about.html]: Here too we don't need caps on custom functions

**Comment by akrantz** [comment]
Incorporated into https://github.com/OfficeDev/Excel-Custom-Functions/pull/55.

**Pull request closed without merging**